hooks/playbooks/ironic_inject_tls.yml

```diff
@@ -0,0 +1,117 @@
+          - '!all'
+          - '!min'
+          - user_gid
+          - user_id
+
+    - name: Create directory
+      ansible.builtin.file:
+        path: "{{ cifmw_basedir }}/artifacts/ironic_ipa_tls/initramfs"
+        state: directory
+        mode: "0755"
+
+    - name: Get internal-ca-bundle and IPA image from k8s resources
+      ansible.builtin.set_fact:
+        internal_ca_bundle: >-
+          {{
+            query(
+              'kubernetes.core.k8s',
+              api_version='v1',
+              kind='Secret',
+              resource_name='combined-ca-bundle',
+              namespace=_namespace
+            ) | first
+              | community.general.json_query('data."internal-ca-bundle.pem"')
+          }}
+        ironic_python_image: >-
+          {{
+            query(
+              'kubernetes.core.k8s',
+              api_version='v1beta1',
+              kind='Ironic',
+              resource_name='ironic',
+              namespace=_namespace
+            ) | first
+              | community.general.json_query('spec.images.ironicPythonAgent')
+          }}
+
+    - name: Write internal-ca-bundle to file
+      ansible.builtin.copy:
+        dest: "{{ cifmw_basedir }}/artifacts/ironic_ipa_tls/internal-ca-bundle.pem"
+        content: "{{ internal_ca_bundle | b64decode }}"
+        mode: "0644"
+
+    - name: Pull and extract ironic-python-ironic
+      containers.podman.podman_container:
+        image: "{{ ironic_python_image }}"
+        name: ironic-python-agent
+        rm: true
+        volumes:
+          - "{{ cifmw_basedir }}/artifacts/ironic_ipa_tls/:/target:Z"
+
+    - name: Install CA bundle in initramfs
+      become: true
+      ansible.builtin.shell: |
+        set -xe -o pipefail
+        zcat {{ cifmw_basedir }}/artifacts/ironic_ipa_tls/ironic-python-agent.initramfs | cpio -idmv
+        cp {{ cifmw_basedir }}/artifacts/ironic_ipa_tls/internal-ca-bundle.pem etc/pki/ca-trust/source/anchors/
+        chroot {{ cifmw_basedir }}/artifacts/ironic_ipa_tls/initramfs << EOF_CHROOT_SCRIPT
+        update-ca-trust
+        EOF_CHROOT_SCRIPT
+        find . | cpio -o -c -R root:root | gzip -9 > {{ cifmw_basedir }}/artifacts/ironic_ipa_tls/tls-ironic-python-agent.initramfs
+      args:
+        chdir: "{{ cifmw_basedir }}/artifacts/ironic_ipa_tls/initramfs"
+
+    - name: Change owner and permissions on tls-ironic-python-agent.initramfs
+      become: true
+      ansible.builtin.file:
+        path: "{{ cifmw_basedir }}/artifacts/ironic_ipa_tls/tls-ironic-python-agent.initramfs"
+        owner: "{{ ansible_user_id }}"
+        group: "{{ ansible_user_gid }}"
+        mode: "0644"
+
+    - name: Recursively delete the content in artifacts/ironic_ipa_tls/initramfs
+      become: true
+      ansible.builtin.file:
+        path: "{{ cifmw_basedir }}/artifacts/ironic_ipa_tls/initramfs/"
+        state: absent
+
+    - name: Create images in openstack and configure ironic node driver info
+      ansible.builtin.shell: |
```

Comment:
ditto - `cifmw.general.ci_script` might be nice

Reply:
Done.